### ISSUE TITLE
Crash in gui/presentation.getBranchDisplayStrings for short branch names

### ISSUE THREAD

**Issue by csinkers** [jesseduffield/lazygit#3217]
When using lazygit on a branch with a two character long name, a panic occurred due to incorrect string slicing. The issue doesn't occur right away so not sure of exact repro steps, but it does tend to occur pretty consistently within a few minutes of use. Edit: Found a way to repro, see below.

Stack trace:
```
Exception has occurred: panic
"runtime error: slice bounds out of range [:3] with length 2"
Stack:
	 3  0x0000000000e7f3d4 in github.com/jesseduffield/lazygit/pkg/gui/presentation.getBranchDisplayStrings
	     at C:/Git/3rdParty/lazygit/pkg/gui/presentation/branches.go:84
	 4  0x0000000000e7ddc6 in github.com/jesseduffield/lazygit/pkg/gui/presentation.GetBranchListDisplayStrings.func1
	     at C:/Git/3rdParty/lazygit/pkg/gui/presentation/branches.go:35
	 5  0x0000000000e8c224 in github.com/samber/lo.Map[go.shape.*uint8,go.shape.[]string]
	     at C:/Git/3rdParty/lazygit/vendor/github.com/samber/lo/slice.go:29
	 6  0x0000000000e7db71 in github.com/jesseduffield/lazygit/pkg/gui/presentation.GetBranchListDisplayStrings
	     at C:/Git/3rdParty/lazygit/pkg/gui/presentation/branches.go:33
	 7  0x0000000000e958c5 in github.com/jesseduffield/lazygit/pkg/gui/context.NewBranchesContext.func3
	     at C:/Git/3rdParty/lazygit/pkg/gui/context/branches_context.go:28
	 8  0x0000000000e99862 in github.com/jesseduffield/lazygit/pkg/gui/context.(*ListRenderer).renderLines
	     at C:/Git/3rdParty/lazygit/pkg/gui/context/list_renderer.go:88
	 9  0x0000000000e990b3 in github.com/jesseduffield/lazygit/pkg/gui/context.(*ListContextTrait).HandleRender
	     at C:/Git/3rdParty/lazygit/pkg/gui/context/list_context_trait.go:88
	10  0x0000000000ebbf97 in github.com/jesseduffield/lazygit/pkg/gui/context.(*BranchesContext).HandleRender
	     at <autogenerated>:1
	11  0x000000000104df7f in github.com/jesseduffield/lazygit/pkg/gui.(*Gui).layout
	     at C:/Git/3rdParty/lazygit/pkg/gui/layout.go:165
	12  0x000000000105c99a in github.com/jesseduffield/lazygit/pkg/gui.(*Gui).layout-fm
	     at <autogenerated>:1
	13  0x0000000000a85ead in github.com/jesseduffield/gocui.ManagerFunc.Layout
	     at C:/Git/3rdParty/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:699
	14  0x0000000000a8949b in github.com/jesseduffield/gocui.(*Gui).flush
	     at C:/Git/3rdParty/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1155
	15  0x0000000000a867f7 in github.com/jesseduffield/gocui.(*Gui).processEvent
	     at C:/Git/3rdParty/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:768
	16  0x0000000000a862cc in github.com/jesseduffield/gocui.(*Gui).MainLoop
	     at C:/Git/3rdParty/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:741
	17  0x000000000104045f in github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run
	     at C:/Git/3rdParty/lazygit/pkg/gui/gui.go:702
	18  0x00000000010408db in github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError.func1
	     at C:/Git/3rdParty/lazygit/pkg/gui/gui.go:708
	19  0x0000000000bc2502 in github.com/jesseduffield/lazygit/pkg/utils.SafeWithError
	     at C:/Git/3rdParty/lazygit/pkg/utils/utils.go:110
	20  0x00000000010407fb in github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError
	     at C:/Git/3rdParty/lazygit/pkg/gui/gui.go:707
	21  0x000000000108fe8c in github.com/jesseduffield/lazygit/pkg/app.(*App).Run
	     at C:/Git/3rdParty/lazygit/pkg/app/app.go:265
	22  0x000000000108d2c9 in github.com/jesseduffield/lazygit/pkg/app.Run
	     at C:/Git/3rdParty/lazygit/pkg/app/app.go:48
	23  0x00000000010918d1 in github.com/jesseduffield/lazygit/pkg/app.Start
	     at C:/Git/3rdParty/lazygit/pkg/app/entry_point.go:150
	24  0x0000000001094e6d in main.main
	     at C:/Git/3rdParty/lazygit/main.go:23
```

**To Reproduce**
Uncertain, but just using lazygit on a branch with a short name tends to crash within 10 mins.
Edit: Found a reliable way to get this crash. When viewing a repo containing a two-character branch name, reduce the width of the terminal window down to a couple of characters. It should crash immediately.

**Version info:**
lazygit built from commit 5c888a0b474ed2fc45c7640b4e73e2bf272f2d3c with VS Code on Windows 10
git version 2.43.0.windows.1

In this case, the relevant local variables in presentation.getBranchDisplayStrings have these values at the point of the crash:

displayName = "sc"
viewWidth = -1
availableWidth = -7
len = 4

The issue seems to be that the viewWidth was -1, but I don't know enough about the rest of the GUI system to tell if this is a valid value and getBranchDisplayStrings should be changed to handle it, or if it's a value that should never be seen and the bug is elsewhere in the GUI code. The particular view that has the width is c.Views().Branches.

**Comment by stefanhaller** [comment]
Thanks for the detailed report. This is a regression caused by #3075.

I couldn't reproduce a case where viewWidth is -1, no idea how this can ever happen; but it could theoretically also happen with positive (but very small) values for viewWidth, and we also have unit tests for this that allow us to check the possible value space. The fix is pretty simple, see #3219.

**Comment by csinkers** [comment]
I haven't seen any issues since 3219 was merged so this looks fixed. Thanks!
